promise.js

```diff
@@ -0,0 +1,48 @@
+/*jslint browser: true, node: true, maxlen: 128 */
+/*global self */
+
+"use strict";
+
+var globalObject, hasPromiseSupport;
+
+function isFunction(x) {
+    return typeof x === "function";
+}
+
+// Seek the global object
+if (global !== undefined) {
+    globalObject = global;
+} else if (window !== undefined && window.document) {
+    globalObject = window;
+} else {
+    globalObject = self;
+}
+
+hasPromiseSupport =
+
+    globalObject.hasOwnProperty("Promise") &&
+
+    // Some of these methods are missing from
+    // Firefox/Chrome experimental implementations
+    globalObject.Promise.hasOwnProperty("resolve") &&
+    globalObject.Promise.hasOwnProperty("reject") &&
+    globalObject.Promise.hasOwnProperty("all") &&
+    globalObject.Promise.hasOwnProperty("race") &&
```

Comment:
Why are you testing for all these methods on `Promise`? You use exactly none of them!

Drop lines 23-30.